File: Leetcode/941. Valid Mountain Array/solution4.py
class Solution:
    def validMountainArray(self, A: List[int]) -> bool:
        if len(A) < 3 or A[0] >= A[1] or A[-2] <= A[-1]: return False
        i = len(A)
        while i > 1 and A[i - 1] < A[i - 2]: i -= 1
        while i > 1 and A[i - 2] < A[i - 1]: i -= 1
        return i == 1

File: Leetcode/1. Two Sum/solution4.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Integer[] map = new Integer[2048];

        for (int i = 0; i < nums.length; i++) {
            int complement = (target - nums[i]) & 2047;

            if (map[complement] != null) {
                return new int[]{map[complement], i};
            }

            map[nums[i] & 2047] = i;
        }

        throw new IllegalArgumentException("No two sum solution");
    }
}